util/usermetric/usermetric.go

```diff
@@ -24,15 +28,18 @@ var vars expvar.Map
 // Note that usermetric are not protected against duplicate
 // metrics name. It is the caller's responsibility to ensure that
 // the name is unique.
-func NewMultiLabelMap[T comparable](name string, promType, helpText string) *metrics.MultiLabelMap[T] {
-	m := &metrics.MultiLabelMap[T]{
+func NewMultiLabelMap[T comparable](m *Registry, name string, promType, helpText string) *metrics.MultiLabelMap[T] {
```

Comment:
Any thoughts on this approach vs `NewMultiLabelMap` being a method on `Registry`?

Reply:
I might be wrong, but my impression is that there is some limitations to when you can have a function on a struct which is generic. I was unable to work out how to do it, here is a goplay with the issue I had: https://play.golang.com/p/8c2nYnE4XJn